storage/rocksdb/rdb_datadic.cc

```diff
@@ -4685,6 +4706,27 @@ bool Rdb_dict_manager::get_cf_flags(const uint32_t &cf_id,
   return found;
 }
 
+void Rdb_dict_manager::get_key_from_index_id(
```

Comment:
That would be the nicest! However dump_index_id is private

Reply:
The places where you reference dump_index_id() should probably be inside the Rdb_dict_manager then. The keys generated for reading/writing the __system__ column family should in Rdb_dict_manager, and you can add an API to return the auto increment value, rather than return the key and require the caller to perform the read.